frontend/app/ui/subscriptions/list/controller.js

```diff
@@ -11,21 +12,52 @@ export default class SubscriptionsListController extends Controller {
   @service intl;
 
   @tracked projects;
+  @tracked searches = {};
+  @tracked sortBy = {};
 
   breadcrumbs = [
     { label: this.intl.t("page.subscriptions.title"), route: "subscriptions" },
     { label: this.intl.t("page.subscriptions.list.title") },
   ];
 
-  @action search(key, query) {
-    const normalized = query.trim().toLowerCase();
+  @action searchFor(key, query) {
+    if (isEmpty(query)) {
+      delete this.searches[key];
+    } else {
+      this.searches[key] = query;
+    }
+
+    this.search();
+  }
+
+  @action limitCritical(event) {
+    const { checked } = event.target;
+
+    if (checked) {
+      this.searches.isTimeAlmostConsumed = "true";
+    } else {
+      delete this.searches.isTimeAlmostConsumed;
+    }
+
+    this.search();
+  }
 
-    this.projects =
-      normalized.length > 0
-        ? this._projects.filter((project) => {
-            return project.get(key).toLowerCase().includes(normalized);
-          })
-        : this._projects;
+  @action search() {
+    let projects = this._projects;
+
+    for (const key in this.searches) {
```

Comment:
I think this whole for can be done as. I think this adheres more to modern js array practises and does not use that much state modification and maybe has less side effect since we dont mutate the `this._projects` on every for loop run.
```js
this.projects = this.searches
  .map(key => String(this.searches[key]).trim().toLowerCase())
  .filter(normalized => !isEmpty(normalized))
  .flatMap(normalized => this._projects.filter(
    (project) => String(project.get(key)).toLowerCase().includes(normalized)
  )
```

Reply:
I did the other suggestion. The chain looses access to the key. I guess you could pass along, say, an array with the key and the normalized value. But that would - IMO - be taking it too far.

And I never mutate `this._projects` and only mutate `this._projects` once at the end of the function outside the loop.

Also, I don't like "modern". Just because something is new doesn't mean that it is better suited for all situations. The `??` operator is a prime example. And "modern" changes over time. With the advent of `async/await` and `for ... of` these "old" coding practices are having a comeback.